### PR TITLE
OpenEnded questions must be of type practice

### PR DESCRIPTION
**Pull request by nathanstitt**
Disable changing the controls when doing so would make question invalid

image

### CODE REVIEW

File: shared/src/model/exercise.js [hunk 1]
```diff
@@ -95,4 +95,15 @@ class Exercise extends BaseModel {
   @computed get isPublishable() {
     return Boolean(!this.isNew && this.validity.valid && !this.published_at);
   }
+
+  @action onQuestionFreeResponseSelected() {
+    if (this.isMultiPart) { return; }
```
Comment (Dantemss) [inline shared/src/model/exercise.js:100]: This return seems odd to me. Are multipart WRQs allowed elsewhere?

File: shared/src/model/exercise/question.js [hunk 2]
```diff
@@ -41,6 +41,16 @@ class ExerciseQuestion extends BaseModel {
 
   @belongsTo({ model: 'exercise' }) exercise;
 
+  @computed get allowedFormatTypes() {
+    if (this.exercise.isSinglePart) {
```
Comment (Dantemss) [inline shared/src/model/exercise/question.js:45]: Same

**Review by Dantemss — COMMENTED** [review]
See comments

File: shared/src/model/exercise.js (hunk 1 above)
Comment (nathanstitt) [reply shared/src/model/exercise.js:100]: yeah it also seemed odd to me but that's exactly what the card read and that's why I asked about practice 😁

**Review by Dantemss — APPROVED** [review]

**Pull request merged**